File: backed-CrewForge/pojo/src/main/java/com/hina/crewforge/pojo/entity/Project.java
package com.hina.crewforge.pojo.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 项目 (sys_project)
 * 用户创建的 AI 编程项目, 一个项目属于一个团队
 * 包含需求 → 澄清 → 规划 → 执行 全过程的产物快照
 */
@Data
@TableName("sys_project")
public class Project {

    /** 确认模式: 全绿灯(Agent自动执行, 无需人工确认) */
    public static final Integer CONFIRM_MODE_GREEN = 0;
    /** 确认模式: 混合(关键步骤人工确认) */
    public static final Integer CONFIRM_MODE_MIXED = 1;
    /** 确认模式: 手动(每一步人工确认) */
    public static final Integer CONFIRM_MODE_MANUAL = 2;

    /** 项目类型: 个人项目(挂在创建人名下, 不属团队) */
    public static final Integer PROJECT_TYPE_PERSONAL = 1;
    /** 项目类型: 团队项目(属于团队工作区) */
    public static final Integer PROJECT_TYPE_TEAM = 2;

    /** 项目ID */
    @TableId(type = IdType.AUTO)
    private Long id;
    /** 所属团队ID(NULL=个人项目) */
    private Long tenantId;
    /** 项目类型: 1-个人项目, 2-团队项目(见常量 PROJECT_TYPE_*) */
    private Integer projectType;
    /** 项目名称 */
    private String name;
    /** 项目描述(原始需求) */
    private String description;
    /** 需求澄清后的结构化文档 */
    private String clarifiedReq;
    /** AI拆分的业务模块列表(JSON) */
    private String businessModules;
    /** 技术栈方案(JSON) */
    private String techStack;
    /** 开发计划(JSON) */
    private String devPlan;
    /** 项目目录树(JSON) */
    private String dirTree;
    /** 状态: draft(草稿)/clarifying(澄清中)/planning(规划中)/executing(执行中)/paused(已暂停)/done(已完成)/failed(失败) */
    private String status;
    /** 确认模式(见下方常量): 0-全绿灯, 1-混合, 2-手动 */
    private Integer confirmMode;
    /** 服务器上项目文件目录 */
    private String projectDir;
    /** 创建人ID */
    private Long createUser;
    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    /** 最后更新时间 */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    /** 逻辑删除: 0-未删除, 1-已删除 */
    @TableLogic
    private Integer deleted;
}

File: fronted-CrewForge/src/types/project.ts
/**
 * 项目相关类型定义（与后端 Entity/API 对齐）
 */

/** 项目状态（与后端 sys_project.status 对应） */
export type ProjectStatus =
  | 'draft'        // 草稿
  | 'clarifying'   // 需求澄清中
  | 'planning'     // 规划中
  | 'executing'    // 执行中
  | 'paused'       // 已暂停
  | 'done'         // 已完成
  | 'failed'       // 失败

/** 确认模式（创建提交时用字符串；后端返回数字 0/1/2） */
export type ConfirmMode = 'green' | 'mixed' | 'manual'

/** 项目（对应后端 ProjectVO） */
export interface Project {
  /** 项目 ID */
  id: number
  /** 项目名称 */
  name: string
  /** 项目描述（原始需求：这个项目要做什么样子的项目） */
  description: string
  /** 项目状态: draft/clarifying/planning/executing/paused/done/failed */
  status: ProjectStatus
  /** 确认模式: 0-全绿灯, 1-混合, 2-手动（后端返回数字） */
  confirmMode: number
  /** 创建人用户 ID */
  createUser: number
  /** 创建时间 */
  createTime: string
  /** 最后更新时间 */
  updateTime: string
  /** 需求澄清后的结构化文档(Markdown)，项目经理澄清后生成 */
  clarifiedReq?: string | null
  /** 业务模块/功能列表（后端是 JSON 字符串，如 ["客户管理","跟进记录"]） */
  businessModules?: string | null
  /** 技术栈列表（后端是 JSON 字符串，如 ["Spring Boot","Vue 3"]，前端解析成数组） */
  techStack?: string | null
  /** 开发计划（后端是 JSON 字符串，架构师生成） */
  devPlan?: string | null
  /** 项目目录树（后端是 JSON 字符串，架构师设计，形如 [{name,type,children}]） */
  dirTree?: string | null
  /** 技术栈标签数组（前端从 techStack 解析后填充） */
  stack?: string[]
  /** 整体进度 0-100（后端暂无此字段，保留兼容旧展示） */
  progress?: number
  /** 文件数（后端暂无此字段，保留兼容旧展示） */
  fileCount?: number
  /** 模块数（后端暂无此字段，保留兼容旧展示） */
  moduleCount?: number
}

/** 新建/更新项目请求（对应后端 ProjectDTO，字段白名单与后端对齐）
 * 创建时必填 name/description/confirmMode；
 * 更新时用 Partial<ProjectCreateDTO> 只传要改的字段 */
export interface ProjectCreateDTO {
  /** 项目名称 */
  name: string
  /** 项目描述（要做什么样子的项目） */
  description: string
  /** 确认模式（前端字符串，提交时经 CONFIRM_MAP 转数字 0/1/2） */
  confirmMode: ConfirmMode
  /** 需求澄清后的结构化文档(Markdown) */
  clarifiedReq?: string
  /** 业务模块/功能列表(JSON 数组字符串)，如 '["客户管理","跟进记录"]' */
  businessModules?: string
  /** 技术栈列表(JSON 数组字符串)，如 '["Spring Boot","Vue 3"]' */
  techStack?: string
  /** 开发计划(JSON 数组字符串) */
  devPlan?: string
  /** 项目目录树(JSON 数组字符串) */
  dirTree?: string
  /** 项目状态: draft/clarifying/planning/executing/paused/done/failed */
  status?: ProjectStatus
}
